File: Problems/Palindrome.cpp
class Solution {
public:
   bool isPalindrome(int x) {
    if (x < 0) return false;
    long y=0;
    long x1=x;
    while (x1>0) {
        y = y*10 + x1%10;
        x1/=10;
    }
    return (y == x);
}
};

File: Problems/Palindrome.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0)
            return false;
        else if(String.valueOf(x).length() == 1)
            return true;
        
        String str = String.valueOf(x);
        
        int i = 0;
        int j = str.length() -1;
        while(j > i){
            if(str.charAt(i) != str.charAt(j))
                return false;
            i++;
            j--;
        }
        return true;
    }
}
